include/zfs_fletcher.h

```diff
@@ -108,6 +108,13 @@ typedef union fletcher_4_ctx {
 #if defined(__aarch64__)
 	zfs_fletcher_aarch64_neon_t aarch64_neon[4];
 #endif
+	}; /* Sure hope anonymous unions work everywhere */
+
+#if defined(_WIN32) && defined(_KERNEL) /* kfpu_begin_ctx() */
+	NTSTATUS saveStatus;
+	XSTATE_SAVE SaveState;
+#endif
+
```

Comment:
I would prefer it if we put these into `zio_abd_checksum_data_t` and put `boolean_t call_kfpu;` into `fletcher_4_ops_t`. The thing that I dislike about changing `fletcher_4_ctx` to a struct and passing it to `kfpu_begin_ctx()` and `kfpu_end_ctx()`. ties those functions to `fletcher_4_ctx`, when they are general purpose functions that should be usable for far more than fletcher4.

I guess another way would be to put these members before the anonymous union, make a structure for these members and use a void pointer to do casting, but I like your `boolean_t call_kfpu;` idea from the other day.

Reply:
This is just a stop-gap until, hopefully, the new implementation @AttilaFueloep  is working on merges. It is a separate commit, so I can easily drop it when that time comes.

Reply:
So there seems to be agreement that the refactoring I suggested yesterday is a good idea? If so, I'll prepare a PR tomorrow.

Reply:
If my vote counts, I'm all for it.

Reply:
It looks good to me.

Reply:
#14600